src/components/subheader/SubHeader.js

```diff
@@ -0,0 +1,75 @@
+// Libraries
+import React from "react";
+
+// components
+import Description from "./components/description";
+import DiscountSection from './components/discountSection'
+import Loader from "../../commonComponents/loader";
+import ErrorHandler from '../../commonComponents/errorHandler'
+
+// readers
+import subheaderReader from './readers/subheaderReader'
+
+// constants
+import { INITIAL_STATE, subheaderDataUrl } from "./constants/subheader.general";
+
+// helpers
+import fetchData from "../../api/fetch";
+
+// css
+import './subHeader.css'
+
+class SubHeader extends React.Component{
+  
+  constructor(props){
+    super(props);
+    this.state = INITIAL_STATE;
+  }
+
+  componentDidMount(){
+    this.getSubheaderData();
+  }
+
+  getSubheaderData(){
+    fetchData(subheaderDataUrl)
+    .then(this.setSubheaderData)
+    .catch(this.handleError)
+    .finally(this.setLoaded)
+  }
+  
+  setSubheaderData = (data) =>{
+    this.setState({ subheaderData : data });
+  }
+
+  handleError = (err) =>{
```

Comment:
even this function name doesn't tell the reader for what purpose you're handling the error, either they have to go through the implementation details or they have to assume it's kinda generic error handling.

Name it more specific